internal/clientsholder/clientsholder.go

```diff
@@ -167,12 +168,20 @@ func createTempFile(prefix string) (file *os.File, err error) {
 	return file, nil
 }
 
+func createByteArrayKubeConfig(kubeConfig *clientcmdapi.Config) ([]byte, error) {
+	yamlBytes, err := clientcmd.Write(*kubeConfig)
+	if err != nil {
+		return []byte{}, fmt.Errorf("failed to generate yaml bytes from kubeconfig: %w", err)
```

Comment:
I guess you can just return `nil` instead of an empty `byte` slice.

Reply:
Done.